### PR TITLE
Build optimized file and wrap in UMD

### PR DESCRIPTION
**Pull request by muffinresearch**
This builds the optimized file with r.js which builds a roll-up of all the modules and uses a tiny require stand-in (almond.js) to provide the amd API internally.

This is then wrapped in some UMD fragments so the result should work in both AMD + common JS and as a global. 

I've used the example apps to check it works. I was thinking we could move them to using requirejs like the tests but this is more like the real usage. I did move them to using the debug builds however.

We'll need to test this. It should mean that fireplace can now consume this as an AMD module. Right now the module exposed is anonymous, if that's a problem we could give it an id.

### CODE REVIEW

**Comment by muffinresearch** [comment]
I should add I've not tried the packaged app yet.

File: Gruntfile.js [hunk 1]
```diff
@@ -166,6 +150,28 @@ module.exports = function(grunt) {
         repo: 'git@github.com:mozilla/fxpay.git'
       },
       src: ['**']
+    },
+
+    requirejs: {
+      debug: {
+        options: {
+          include: ['../../node_modules/almond/almond'],
```
Comment (kumar303) [inline Gruntfile.js]: I think it would be a good idea if this used ``__dirname``

**Comment by kumar303** [comment]
I only see the debug file. Is the compressed ``fxpay.min.js`` coming in another patch?

**Comment by kumar303** [comment]
This is a time when I wish our usage docs were in the repo. We'll need to remember to update the MDN docs showing how to include the fxpay module when this lands.

File: Gruntfile.js (hunk 1 above)
Comment (kumar303) [inline Gruntfile.js]: and the other relative paths

**Comment by kumar303** [comment]
r+wc

File: Gruntfile.js (hunk 1 above)
Comment (kumar303) [inline Gruntfile.js]: but I guess all other paths in here are relative?

**Comment by kumar303** [comment]
it would be cool if the ``compress`` task reported an informative error if you run it before running npm install. That is, before the almond module has been installed.

**Comment by muffinresearch** [comment]
> it would be cool if the compress task reported an informative error if you run it before running npm install. That is, before the almond module has been installed.

Done

**Pull request merged**